config/config.go

```diff
@@ -144,6 +145,7 @@ func (c Configuration) New(
 	tracerOptions := []jaeger.TracerOption{
 		jaeger.TracerOptions.Metrics(tracerMetrics),
 		jaeger.TracerOptions.Logger(opts.logger),
+		jaeger.TracerOptions.TracerHeaderKeys(c.Headers),
```

Comment:
node that these custom headers are considered a config option to the user. However, internally, we don't want to be invasive to the codebase and treat them as a configuration item throughout. Hence, while they are considered a configuration item to the user, internally, they are considered an option.